src/lib.rs

```diff
@@ -106,7 +141,7 @@ impl Guess<'_> {
     pub fn matches(&self, word: &str) -> bool {
         // if guess G gives mask C against answer A, then
         // guess A should also give mask C against answer G.
-        Correctness::compute(word, &self.word) == self.mask
+        Correctness::check(word, &self.word, &self.mask)
```

Comment:
Actually, at this point, we can really just place the body of `Correctness::check` directly in here. No need to force it to be a static method on `Correctness` :+1: 

Also, the comment above is no longer accurate — that was trying to explain why doing it the old way yielded the right answer.

Reply:
I moved it there now, and wrote more up to date comment. I also mentioned the original check in the comment, as I think it shows the property that is computed here quite clearly.